controllers/reconcilers.go

```diff
@@ -127,6 +135,70 @@ func (r *InstanceGroupReconciler) configMapReconciler(obj handler.MapObject) []c
 	return nil
 }
 
+func (r *InstanceGroupReconciler) namespaceReconciler(obj handler.MapObject) []ctrl.Request {
+	var (
+		name = obj.Meta.GetName()
+	)
+
+	ctrl.Log.Info("namespace watch event", "object", name)
+
+	namespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      name,
+	}
+
+	r.NamespacesLock.Lock()
+	defer r.NamespacesLock.Unlock()
+
+	err := r.Get(context.Background(), namespacedName, obj.Object)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			r.Log.Info("namespace deleted", "object", name)
```

Comment:
You should test this, if your watch only triggers on deletionTimestamp being set then your code here will likely never be triggered.

Reply:
Tested this, it correctly detects deletion. a watch event is triggered on delete, and in that case the object is no longer found.